packages/main/test/pages/TreeDragAndDrop.html

```diff
@@ -0,0 +1,126 @@
+	<title>Tree Drag and Drop</title>
+
+	<script src="%VITE_BUNDLE_PATH%" type="module"></script>
+	<link rel="stylesheet" type="text/css" href="./styles/List.css">
+</head>
+
+<body class="tree1auto">
+	<ui5-checkbox id="reorder" text="Movable items" checked></ui5-checkbox>
+	<ui5-checkbox id="density" text="Compact density"></ui5-checkbox>
+	<a id="link" href="http://sap.com" draggable="true">http://sap.com</a>
+
+	<section>
+		<h2>Drag and drop</h2>
+		<ui5-tree id="tree" no-data-text="No data" mode="MultiSelect" accessible-name="Tree with accessibleName">
+	
+			<ui5-tree-item movable expanded text="Tree 1" icon="paste" additional-text="Available" indeterminate selected additional-text-state="Information" accessible-name="Tree item with accessibleName">
+				<ui5-title slot="content"> 
+					<ui5-label>Tree 1</ui5-label>
+					<ui5-label>Tree 1</ui5-label>
+				</ui5-title>
+	
+				<ui5-tree-item movable expanded text="Tree 1.1" additional-text="Re-stock" additional-text-state="Error" indeterminate selected>
+					<ui5-tree-item movable text="Tree 1.1.1" additional-text="Required" additional-text-state="Warning" selected></ui5-tree-item>
+					<ui5-tree-item movable text="Tree 1.1.2" additional-text="Available" additional-text-state="Success"></ui5-tree-item>
+				</ui5-tree-item>
+			</ui5-tree-item>
+	
+			<ui5-tree-item movable text="Tree 2" icon="copy">
+				<ui5-tree-item movable id="firstCollapsedItem" text="Tree 2.1">
+					<ui5-tree-item movable text="Tree 2.1.1"></ui5-tree-item>
+					<ui5-tree-item movable text="Tree 2.1.2">
+						<ui5-tree-item movable text="Tree 2.1.2.1"></ui5-tree-item>
+						<ui5-tree-item movable text="Tree 2.1.2.2"></ui5-tree-item>
+						<ui5-tree-item movable text="Tree 2.1.2.3"></ui5-tree-item>
+						<ui5-tree-item movable text="Tree 2.1.2.5"></ui5-tree-item>
+					</ui5-tree-item>
+				</ui5-tree-item>
+				<ui5-tree-item movable text="Tree 2.2"></ui5-tree-item>
+			</ui5-tree-item>
+	
+			<ui5-tree-item movable data-allows-nesting expanded text="Tree 3 (no icon) ALLOWS NESTING">
+			</ui5-tree-item>
+	
+		</ui5-tree>
+	</section>
+
+	<!-- <section class="largeTopMargin">
+		<ui5-list id="listDnd2" header-text="List 2">
+			<ui5-li id="bg2" movable>2. Bulgaria</ui5-li>
+			<ui5-li id="de2" movable data-allows-nesting>2. Germany (Allows nesting)</ui5-li>
+			<ui5-li id="es2" movable>2. Spain</ui5-li>
+		</ui5-list>
+	</section> -->
+
+	<script>
+		const tree1 = document.getElementById("tree");
+		const handleBeforeItemMove = (e) => {
+			const { destination, source } = e.detail;
+
+			if (destination.placement === "Before" || destination.placement === "After") {
+				e.preventDefault();
+			}
+
+			if (destination.placement === "On" && "allowsNesting" in destination.element.dataset) {
+				e.preventDefault();
+			}
+
+			console.log(`Moving "${source.element.id}" ${destination.placement.toLowerCase()} "${destination.element.id}"`);
```

Comment:
Items don't have `id`s. Rather add id to all of them, or use `source.element.text` and `destination.element.text`